conda/models/index_record.py

```diff
@@ -81,8 +61,10 @@ class IndexRecord(DictSafeMixin, ImmutableEntity):  # rename to IndexRecord
     url = StringField(required=False, nullable=True)
     auth = StringField(required=False, nullable=True)
 
-    files = ListField(string_types, default=(), required=False)
-    link = ComposableField(Link, required=False)
-
     with_features_depends = MapField(required=False)
     preferred_env = StringField(default=None, required=False, nullable=True)
+
+
+class LinkedPackageRecord(IndexRecord):
+    files = ListField(string_types, default=(), required=False)
+    link = ComposableField(Link, required=False)
```

Comment:
To preserved the behavior for what `linked_data` returns, I've finally sub-classed `IndexRecord` and now moved these two fields.  I've had a `TODO` for this for a long time.